omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -96,11 +99,17 @@
             if(Context.hasPrivilege("Manage OWA")){
                 String message;
                 HttpSession session = request.getSession();
+                String fileName = file.getOriginalFilename();
+                File uploadedFile = new File(file.getOriginalFilename());
+                file.transferTo(uploadedFile);
+                FileInputStream fis = new FileInputStream(uploadedFile);
+                ZipInputStream zis = new ZipInputStream(fis);
+                boolean isZip = zis.getNextEntry() != null;
+                fis.close(); 
+                zis.close();
                 if (!file.isEmpty()) {
-                    String fileName = file.getOriginalFilename();
-                    File uploadedFile = new File(file.getOriginalFilename());
-                    file.transferTo(uploadedFile);
-                    try (ZipFile zip = new ZipFile(uploadedFile)) {
+                    if (isZip) {
+                    	ZipFile zip = new ZipFile(uploadedFile);
```

Comment:
@Vincentes No need to make so many changes to solve the error of "response is already committed". Please revert it to original code and add else block to if(zip.size == 0). 
So after reverting to original code add else block at line 110 and end else block after line 123 i.e. above catch block.